lib/ansible/modules/network/mlnxos/mlnxos_bgp.py

```diff
@@ -0,0 +1,197 @@
+#!/usr/bin/python
+#
+# Copyright: Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = """
+---
+module: mlnxos_bgp
+version_added: "2.5"
+author: "Samer Deeb (@samerd)"
+short_description: Configures BGP on Mellanox MLNX-OS network devices
+description:
+  - This module provides declarative management of BGP router and neighbors
+    on Mellanox MLNX-OS network devices.
+notes:
+  - Tested on MLNX-OS 3.6.4000
+options:
+  as_number:
+    description:
+      - Local AS number.
+    required: true
+  router_id:
+    description:
+      - Router IP address. Required if I(state=present).
+  neighbors:
```

Comment:
Should this be required (if so please update argspec)

Reply:
Thanks @gundalow
neighbors is required only if state is 'present'
Shall I update the documentation?

Reply:
Yup, add that in the `description`.

Reply:
Yup, add that in the `description`.

Reply:
Done.
Thanks a lot for the notice.